src/test/java/org/commcare/formplayer/tests/RestoreFactoryTest.java

```diff
@@ -173,6 +200,50 @@ public void testGetRequestHeaders() {
         );
     }
 
+    @Test
+    public void testGetRequestHeaders_HmacAuth() throws Exception {
+        mockHmacRequest();
+        restoreFactorySpy.configure(domain, "case_id", null);
+        String requestPath = "/a/restore-domain/case_migrations/restore/case_id_123/";
+        HttpHeaders headers = restoreFactorySpy.getRequestHeaders(new URI("http://localhost:8000" + requestPath));
+        assertEquals(4, headers.size());
+        validateHeaders(headers, Arrays.asList(
+                hasEntry("X-MAC-DIGEST", singletonList(RequestUtils.getHmac(formplayerAuthKey, requestPath))),
+                hasEntry("X-OpenRosa-Version", singletonList("3.0")),
+                hasEntry("X-OpenRosa-DeviceId", singletonList("WebAppsLogin")),
+                hasEntry(equalTo("X-CommCareHQ-Origin-Token"), new ValueIsUUID()))
+        );
+    }
+
+    @Test
+    public void testGetRequestHeaders_HmacAuth_UrlWithQuery() throws Exception {
+        mockHmacRequest();
+        restoreFactorySpy.configure(domain, "case_id", null);
+        String requestPath = "/a/restore-domain/case_migrations/restore/case_id_123/?query_param=true";
+        HttpHeaders headers = restoreFactorySpy.getRequestHeaders(new URI("http://localhost:8000" + requestPath));
+        assertEquals(4, headers.size());
+        validateHeaders(headers, Arrays.asList(
+                hasEntry("X-MAC-DIGEST", singletonList(RequestUtils.getHmac(formplayerAuthKey, requestPath))),
+                hasEntry("X-OpenRosa-Version", singletonList("3.0")),
+                hasEntry("X-OpenRosa-DeviceId", singletonList("WebAppsLogin")),
+                hasEntry(equalTo("X-CommCareHQ-Origin-Token"), new ValueIsUUID()))
+        );
+    }
+
+    @Test
+    public void testGetRequestHeaders_HmacAuthRequestWithUserDetails() throws Exception {
```

Comment:
curious how is this test passsing the user details ?

Reply:
Ah, renamed the test to `testGetRequestHeaders_UseHmacAuthEvenIfHqAuthPresent`

The hq auth is set in the setup method: https://github.com/dimagi/formplayer/pull/1060/files#diff-82c4b4770fc4eb29704a291fad1ff9e3f3cb061d8ff1044b24d70b756ffbaa2eR74